mcp2210-core.c

```diff
@@ -709,6 +712,358 @@ static inline int eeprom_read_complete(struct mcp2210_cmd *cmd_head,
 }
 #endif /* CONFIG_MCP2210_CREEK */
 
+#ifdef CONFIG_OF
+/* --------------------------------------------------------------------------*/
+/**
+ * @brief This function is used to parse the device tree node for mcp2210
+ *        if available and based on that configure the mcp2210 bridge.
+ *
+ * @Param dev mcp2210 device struct pointer
+ *
+ * @Returns  0 on success, negative values on error
+ */
+/* ----------------------------------------------------------------------------*/
+static int mcp2210_parse_dts(struct mcp2210_device *dev)
+{
+    struct device_node *np;
+    struct mcp2210_board_config board_config;
+    struct mcp2210_chip_settings chip_settings;
+    struct mcp2210_chip_settings powerup_chip_settings;
+    struct mcp2210_spi_xfer_settings spi_settings;
+    struct mcp2210_spi_xfer_settings powerup_spi_settings;
+    struct mcp2210_usb_key_params usb_key_params;
+    struct mcp2210_ioctl_data *data;
+    struct mcp2210_ioctl_data_config *cfg;
+    size_t struct_size = IOCTL_DATA_CONFIG_SIZE;
+    size_t strings_size = 0;
+    enum mcp2210_ioctl_cmd ioctl_cmd = MCP2210_IOCTL_CONFIG_SET;
+    size_t result_size;
+    struct ioctl_result *result;
+    u8 value;
+    u8 have_board_config = 0;
+    int ret = 0;
+    u32 i;
+
+    /* Initialize the structures */
+    memset (&board_config, 0, sizeof(board_config));
+    memset (&chip_settings, 0, sizeof(chip_settings));
+    memset (&powerup_chip_settings, 0, sizeof(powerup_chip_settings));
+    memset (&spi_settings, 0, sizeof(spi_settings));
+    memset (&powerup_spi_settings, 0, sizeof(powerup_spi_settings));
+    memset (&usb_key_params, 0, sizeof(usb_key_params));
+
+    /* Determine if a device tree entry is available for this driver. */
+    np = of_find_compatible_node(NULL, NULL, "microchip,mcp2210");
+    if(np)
+    {
```

Comment:
Rather than indent a large section of code, just test for failure and exit here.